src/tests.zig

```diff
@@ -2184,3 +2184,68 @@ test "compile and run bytecode" {
     // produced bytecode in text format, but the API doesn't support it.
     try testing.expect(bc1.len < bc2.len);
 }
+
+test "userdata dtor" {
+    if (ziglua.lang != .luau) return;
+    var gc_hits: i32 = 0;
+
+    const Data = struct {
+        gc_hits_ptr: *i32,
+
+        pub fn dtor(udata: *anyopaque) void {
+            const self: *@This() = @alignCast(@ptrCast(udata));
+            self.gc_hits_ptr.* = self.gc_hits_ptr.* + 1;
+        }
+    };
+
+    // create a Lua-owned pointer to a Data, configure Data with a destructor.
+    {
+        var lua = try Lua.init(testing.allocator);
+        defer lua.deinit(); // forces dtors to be called at the latest
+
+        var data = lua.newUserdataDtor(Data, ziglua.wrap(Data.dtor));
+        data.gc_hits_ptr = &gc_hits;
+        try expectEqual(@as(*const anyopaque, @ptrCast(data)), try lua.toPointer(1));
+        try testing.expectEqual(@as(i32, 0), gc_hits);
```

Comment:
I can do a clean up pass and resubmit.  I can fix the same from the other change that was merged, in case it also has these cases.  I think I had see the testing.expectEqual @as change but had not internalized that yet.  It was pain, I'm glad it's fixed.

Reply:
Added another commit that cleans this up.  Not sure if you'd prefer a squashed commit, thought this just easier to review.

Reply:
No need to squash, thanks!